### DISCUSSION TITLE
[Run issue]: Installing on WSL2 - Fix to libQt5Core.so.5 shared library issue

### DISCUSSION THREAD

**Discussion by AndrewFielding** [OpenTOPAS/OpenTOPAS#27] [Run-Problem] [answered]
### Contact Details

andrew.fielding2010@gmail.com

### What was the issue?

This may be useful information for anyone trying to install on WSL2. I followed the instructions and when I tried to run topas with an example I got an error that the libQt5Core.so.5 shared library couldn't be found.

Changing the directory in the Geant4 build (Step 7)  to -DCMAKE_PREFIX_PATH=/usr/lib/x86_64-linux-gnu/ didn't help. I found I needed another fix that is apparently well known, its a WSL issue (https://superuser.com/questions/1347723/arch-on-wsl-libqt5core-so-5-not-found-despite-being-installed). 

You need to run the following command at step 7:

sudo strip --remove-section=.note.ABI-tag /usr/lib/x86_64-linux-gnu/libQt5Core.so.5

This appears to fix the issue (for me) when installing on WSL.

cheers,

Andrew

### TOPAS category

Geometry

### Version

4.0

### What Operative System (OS) are you running on?

Windows (WSL)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by omarRgg** [comment]
I have installed OpenTOPAS on Windows 11 WSL running Ubuntu 20.04 and Windows 10 running Debian 12, I had troubles with QT both times and had to erase Geant4, checked the installation of the libraries on Step 1 and QT on step 5 and re installed G4. That solved the issues but I’m not completely sure what specific step was the solution. Also, there was an problem with one of them that had previously Installed Python Anaconda, the error stated that a file interfered with the QT access and the OpenTOPAS installation failed, I had to erase Anaconda first and restart the installation. Tanks for the information, I will try it next time I have to install OpenTOPAS on a WSL system!

**Comment by JNaokiDKondo** [comment]
Dear Andrew,
Thanks for pointing us towards this issue. We will be releasing later this week a section for Quick Installation of OpenTOPAS on ReadTheDocs. This guide should guide you trough the setting up of your QT libraries.

-Naoki.

**Comment by JNaokiDKondo** [comment, marked as answer]
Hello again,
There is now a new section in the OpenTOPAS ReadTheDocs that explains how to install OpenTOPAS in WSL (https://opentopas.readthedocs.io/en/latest/getting-started/WSL.html#).
Step 7 explains how to install Geant4 which includes the QT directory. It seems that for WSL, the directory is not in the same path as in a pure Ubuntu distribution.

- Naoki.

**Reply by thanhtaiphys** [reply]
@JNaokiDKondo Hi Naoki, I encountered an issue during Step 8 of the OpenTOPAS installation process. Specifically, when I ran the command: " ~/Applications/TOPAS/OpenTOPAS-install/bin/topas"

I received the following error message:

/home/t/Applications/TOPAS/OpenTOPAS-install/bin/topas: error while loading shared libraries: libQt5Core.so.5: cannot open shared object file: No such file or directory

I have attempted to install OpenTOPAS on WLS Ubuntu versions 24.04, 22.04, and 20.04, but I encountered the same error at this step in each case.
Could you please provide any guidance or suggestions on how to resolve this issue?

**Reply by JNaokiDKondo** [reply]
Hello @thanhtaiphys,
Please take a look at this question https://github.com/OpenTOPAS/OpenTOPAS/discussions/34
Most probably your "QT_QPA_PLATFORM_PLUGIN_PATH" environment variable is causing the issue. Try deleting this environment variable and try again.

Cheers,
Naoki.

**Reply by thanhtaiphys** [reply]
Hi,

I deleted "QT_QPA_PLATFORM_PLUGIN_PATH" by using command line "unset QT_QPA_PLATFORM_PLUGIN_PATH" but it still doesn't work

![image](https://github.com/user-attachments/assets/36071062-c8a4-4b61-8f56-a74f1eab7482)

**Reply by thanhtaiphys** [reply]
@JNaokiDKondo  I followed the installation instructions from steps 1 to 8, and everything went well except for the command ~/Applications/TOPAS/OpenTOPAS-install/bin/topas.
